main.go

```diff
@@ -103,9 +104,11 @@ func main() {
 	}
 
 	// initialize kubernetes client, if needed
-	if err := kubernetes.InitClient(); err != nil {
-		setupLog.Error(err, "Failed to initialize kubernetes client")
-		os.Exit(1)
+	if kubernetesEnvironmentNeeded(cfg) {
```

Comment:
Can we call this method `shouldSetupKubeClient`?

Reply:
Sure